[hotfix][docs] Update release notes for 1.13 with state backend migra… (#16448)
apache/flink: docs/content.zh/release-notes/flink-1.13.md

The constructors of `FsStateBackend` and `MemoryStateBackend` that take a flag for sync/async
snapshots are kept for API compatibility, but the flags are ignored now.

##### [FLINK-19463](https://issues.apache.org/jira/browse/FLINK-19463)

Flink has always separated local state storage from fault tolerance.

> Keyed state is maintained locally in state backends, either on the JVM heap or embedded RocksDB instances.

Reviewer: ```suggestion
Keyed state is maintained locally in state backends, either on the JVM heap or in embedded RocksDB instances.
```

> Fault tolerance comes from checkpoints and savepoints - periodic snapshots of the Jobs internal state to some durable file system - such as Amazon S3 or HDFS. 

Reviewer: ```suggestion
Fault tolerance comes from checkpoints and savepoints - periodic snapshots of a job's internal state to some durable file system - such as Amazon S3 or HDFS. 
```

Historically, Flink's `StateBackend` interface intermixed these concepts in a way that confused many users. 
In 1.13, checkpointing configurations have been extracted into their own interface, `CheckpointStorage`. 

This change does not affect the runtime behavior and simply provides a better mental model to users. 
Pipelines can be updated to use the new the new abstractions without losing state, consistency, or change in semantics. 

> Please follow the [migraiton guide](https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/ops/state/state_backends/#migrating-from-legacy-backends) or JavaDoc on the deprecated state backend classes - `MemoryStateBackend`, `FsStateBackend` and `RocksDBStateBackend` for migration details. 

Reviewer: ```suggestion
Please follow the [migration guide](https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/ops/state/state_backends/#migrating-from-legacy-backends) or the JavaDoc on the deprecated state backend classes - `MemoryStateBackend`, `FsStateBackend` and `RocksDBStateBackend` for migration details. 
```
